packages/@react-spectrum/label/src/StaticField.tsx

```diff
@@ -0,0 +1,180 @@
+
+          {value instanceof Date && 
+            <FormattedDate value={value as DateTimeValue} formatOptions={formatOptions as Intl.DateTimeFormatOptions} />}
+          
+          {value instanceof Time && 
+            <FormattedDate value={value as DateTimeValue} formatOptions={formatOptions as Intl.DateTimeFormatOptions} />}
+
+          {typeof value === 'string' && 
+            props.value}
+        </span>
+      </Label>
+    </div>
+  );
+}
+
+function FormattedStringList<T extends string[]>(props: StringListProps<T>) {
+  let stringFormatter = useListFormatter(props.formatOptions); 
+
+  return (
+    stringFormatter.format(props.value)
+  );
+}
+
+function FormattedNumber<T extends NumberValue>(props: NumberProps<T>) {
+  let numberFormatter = useNumberFormatter(props.formatOptions); 
+
+  let value = props.value;
+
+  if (typeof value === 'object' && typeof (value as RangeValue<NumberValue>).start === 'number' && typeof (value as RangeValue<NumberValue>).end === 'number') {
+    console.log('RangeValue<NumberValue>');
+    return <>{numberFormatter.formatRange(value.start, value.end)}</>;
+  }
+
+  return <>{numberFormatter.format(value as number)}</>;
+}
+
+function FormattedDateRange<T extends DateTimeValue>(props: DateProps<T>) {
+  let dateFormatter = useDateFormatter(props.formatOptions);
+  let value = props.value;
+  let timeZone = dateFormatter.resolvedOptions().timeZone || getLocalTimeZone();
+  let start;
+  let end;
+
+  if (typeof value === 'object' && 'start' in value) {
+    start = value.start;
+    end = value.end;
+
+    if (!(start instanceof Date) && !(end instanceof Date)) {
+      if (start instanceof Time && end instanceof Time) {
+        start = dateFormatter.format(convertValue(start).toDate(timeZone));
+        end = dateFormatter.format(convertValue(end).toDate(timeZone));
+      } else if (!(start instanceof Time) && !(end instanceof Time)) {
+        start = start.toDate(timeZone);
+        end = end.toDate(timeZone);
+      }
+    } 
+  }
+  
+  return <>{dateFormatter.formatRange(start, end)}</>;
+}
+
+function FormattedDate<T extends DateTimeValue>(props: DateProps<T>) {
+  let dateFormatter = useDateFormatter(props.formatOptions);
+  let value = props.value;
+  let timeZone = dateFormatter.resolvedOptions().timeZone || getLocalTimeZone();
+
+  if (typeof value === 'object' && 'timeZone' in value) {
+    console.log(value.toDate());
+    return <>{dateFormatter.format(value.toDate())}</>;
+  }
+
+  if (typeof value === 'object' && 'calendar' in value) {
+    return <>{dateFormatter.format(value.toDate(timeZone))}</>;
+  }
+
+  if (value instanceof Time) {    
+    return <>{dateFormatter.format(convertValue(value).toDate(timeZone))}</>;
+  }
+
+  return <>{dateFormatter.format(value as Date)}</>;
+}
+
+function convertValue(value: TimeValue, date: DateValue = today(getLocalTimeZone())) {
```

Comment:
If we don't use the `date` parameter, then we can just get rid of it and initialize it inside the function instead